src/main/java/org/broadinstitute/hellbender/tools/HaplotypeCallerSpark.java

```diff
@@ -267,13 +293,24 @@ private void writeVariants(JavaRDD<VariantContext> variants) {
             final Broadcast<ReferenceMultiSource> reference,
             final Broadcast<HaplotypeCallerArgumentCollection> hcArgsBroadcast,
             final ShardingArgumentCollection assemblyArgs,
-            final SAMFileHeader header) {
+            final SAMFileHeader header,
+            final Broadcast<VariantAnnotatorEngine> annotatorEngineBroadcast) {
```

Comment:
I tried broadcasting the engine, but it's not threadsafe, and I hit a lot of issues because of this. The reason to broadcast the `VariantAnnotatorEngine` is because it scans the classpath, which makes it expensive to create anew on each executor - so broadcasting makes sense.